sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/json/JsonSuite.scala

```diff
@@ -66,7 +66,11 @@ class JsonSuite extends QueryTest with SharedSQLContext with TestJsonData {
 
       val dummyOption = new JSONOptions(Map.empty[String, String], "GMT")
       val dummySchema = StructType(Seq.empty)
-      val parser = new JacksonParser(dummySchema, dummyOption, allowArrayAsStructs = true)
+      val parser = new JacksonParser(
+        dummySchema,
+        dummyOption,
+        allowArrayAsStructs = true,
+        skipInputWithoutTokens = true)
```

Comment:
Shall we have a test coverage for both `true` and `false`? In case of `false`, we should not skip the rows, right?

Reply:
I guess it's handled by `from_json` tests. If not, let's add a new UT here.

Reply:
Both cases are covered already, for example:
- `skipInputWithoutTokens == false` https://github.com/apache/spark/blob/3238e3d1c0d9be5c43a72705e18afbbb4c512e15/sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/expressions/JsonExpressionsSuite.scala#L547-L553
- `skipInputWithoutTokens == true` https://github.com/apache/spark/pull/23325/files#diff-fde14032b0e6ef8086461edf79a27c5dR2520

Reply:
Thank you for clarifying that, @MaxGekk and @cloud-fan .